### DISCUSSION TITLE
class components using `t` macro? way forward?

### DISCUSSION THREAD

**Discussion by cezarneaga** [lingui/js-lingui#912] [Q&A] [answered]
Hi everyone,
without the `<I18N>` render-prop component how can we use the `t` macro in a class component?

thanks,
Cezar

**Comment by cezarneaga** [comment, marked as answer]
i answer myself like a rubber duck that i am:
by using the `withI18n` HoC :)

perhaps a quick mention in the migrate to v3 post would be good. should i PR it @tricoder42  ?

cheers,
C

**Reply by semoal** [reply]
If you use V3 you can use directly t macro without i18n._(t``)

**Reply by cezarneaga** [reply]
f.......k. i am so stupid. this piece of info i missed entirely. i have been calling the `useLingui()` everywhere. thanks @semoal!

**Reply by semoal** [reply]
Glad to help =)
